### PR TITLE
IA-4174: Remove defunct WSM APIs from Leo

### PR DESCRIPTION
**Pull request by rtitle**
Context in: https://broadworkbench.atlassian.net/browse/IA-4174

I will test on a BEE, first to replicate the issue on dev, then verify this fix.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/WsmDao.scala [hunk 1]
```diff
@@ -33,10 +31,6 @@ import java.time.ZonedDateTime
 import java.util.UUID
 
 trait WsmDao[F[_]] {
-  def createIp(request: CreateIpRequest, authorization: Authorization)(implicit
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/WsmDao.scala:36]: I deleted the methods/models from `WsmDao` around IP, network, and relay namespace which are not used anymore.

Side note: I would _really_ like if Leo migrated to using published Java client libraries for WSM (and Sam) in the future. That would reduce boilerplate code and surface for potential contract-related bugs.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala [hunk 2]
```diff
@@ -308,6 +308,16 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
         .raiseError[Unit](RuntimeNotFoundException(runtime.cloudContext, runtimeName, "permission denied"))
         .whenA(!hasPermission)
 
+      // Query WSM for Landing Zone resources
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala:311]: `DeleteAzureRuntimeMessage` now needs `LandingZoneResources`, since it needs the relay namespace in order to delete the hybrid connection. This is a similar pattern to `CreateAzureRuntimeMessage`.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoMetricsMonitor.scala [hunk 3]
```diff
@@ -115,7 +115,8 @@ class LeoMetricsMonitor[F[_]](config: LeoMetricsMonitorConfig,
       c <- r.images.filter(i => imageTypes.contains(i.imageType)).headOption
     } yield Map(
       RuntimeStatusMetric(r.cloudContext.cloudProvider,
-                          c,
+                          c.imageType,
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoMetricsMonitor.scala:118]: This is an unrelated metrics fix, which I also happened to notice since the release on 3/28. Storing `RuntimeImage` in the `RuntimeStatusMetric` and `RuntimeHealthMetric` case classes was breaking aggregation because it holds a timestamp field. Changed that type to just contain the fields we care about for metrics (imageType and imageUrl).

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/WsmDao.scala (hunk 1 above)
Comment (LizBaldo) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/WsmDao.scala:36]: That is a very good point, I will create a tech debt ticket for us to address that

Comment (LizBaldo) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/WsmDao.scala:36]: https://broadworkbench.atlassian.net/browse/IA-4175

**Review by LizBaldo — APPROVED** [review]

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala [hunk 4]
```diff
@@ -585,12 +577,12 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
...
         case x: CloudContext.Azure => F.pure(x.value)
       }
-      _ <- relayNamespaceOpt.traverse(ns =>
-        azureRelay.deleteRelayHybridConnection(
-          ns,
-          RelayHybridConnectionName(runtime.runtimeName.asString),
-          cloudContext
-        )
+
+      // Delete hybrid connection for this VM
```
Comment (aherbst-broad) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala:579]: This is a good catch, explains why the connections were leaking.

Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala:579]: Yup, trying to verify in a BEE that they are being deleted now

**Comment by rtitle** [comment]
Testing notes:

First I reproduced the error in my BEE with dev code: 
image

It works with the Leo fix:
image


Also verified the relay hybrid connection now gets deleted upon VM deletion. I think this partially resolves https://broadworkbench.atlassian.net/browse/IA-4097, will comment in that ticket.

**Comment by rtitle** [comment]
Metrics look good too -- will plan to merge after green tests.

**Comment by rtitle** [comment]
RStudioSpec failed which is a known flake. Going to merge this.
